### PR TITLE
2019/kw19/codecov

### PR DESCRIPTION
**Pull request by Tirokk**
> [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-08T11:52:55Z" title="Wednesday, May 8th 2019, 1:52:55 pm +02:00">May 8, 2019</time>_
_Merged <time datetime="2019-05-08T15:24:09Z" title="Wednesday, May 8th 2019, 5:24:09 pm +02:00">May 8, 2019</time>_
---

## Pullrequest

Codecov for coverage reports

### Issues

- [X] does not yet fi-x https://github.com/Human-Connection/Human-Connection/issues/469

### Todo

- [X] None

### CODE REVIEW

File: docker-compose.travis.yml [hunk 1]
```diff
@@ -11,13 +11,18 @@ services:
...
+    #/nitro-web
+       - ./webapp/coverage:/nitro-web/coverage
     environment:
       - GRAPHQL_URI=http://backend:4000
   backend:
     image: humanconnection/nitro-backend:builder
     build:
       context: backend
       target: builder
+    volumes:
```
Comment (Tirokk) [inline docker-compose.travis.yml:24]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-08T12:13:33Z" title="Wednesday, May 8th 2019, 2:13:33 pm +02:00">May 8, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -11,13 +11,17 @@ services:
     build:
       context: webapp
       target: build-and-test
+    volumes:
+       - coverage:webapp/coverage
     environment:
       - GRAPHQL_URI=http://backend:4000
   backend:
     image: humanconnection/nitro-backend:builder
     build:
       context: backend
       target: builder
+    volumes:
+       - coverage:backend/coverage
```


I'm curious if Travis CI will crash after this. When I tried to add volumes a couple of months ago, Travis could not clean up after this. Reason: The Travis user running the `.travis.yml` is (was?) different from the user of the `docker` daemon. Thus, a file created on the host by `docker-compose` would have a different UNIX file permission. The user on the host machine that tries to clean up after the build would have not enough permissions to remove the created files.

**Comment by Tirokk** [comment]
> [codecov[bot]](https://github.com/apps/codecov) **Authored by [codecov[bot]](https://github.com/apps/codecov)**
_<time datetime="2019-05-08T13:36:31Z" title="Wednesday, May 8th 2019, 3:36:31 pm +02:00">May 8, 2019</time>_

---


# [Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/569?src=pr&el=h1) Report
> :exclamation: No coverage uploaded for pull request base (`master@5a60fa6`). [Click here to learn what that means](https://docs.codecov.io/docs/error-reference#section-missing-base-commit).
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/Human-Connection/Human-Connection/pull/569?src=pr&el=tree)

```diff
@@            Coverage Diff            @@
##             master     #569   +/-   ##
=========================================
  Coverage          ?   58.56%           
=========================================
  Files             ?       41           
  Lines             ?      613           
  Branches          ?       58           
=========================================
  Hits              ?      359           
  Misses            ?      234           
  Partials          ?       20
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/569?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/569?src=pr&el=footer). Last update [5a60fa6...f63ed9d](https://codecov.io/gh/Human-Connection/Human-Connection/pull/569?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by Tirokk** [comment]
> [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-08T13:55:47Z" title="Wednesday, May 8th 2019, 3:55:47 pm +02:00">May 8, 2019</time>_

---


Once this has passed, I would like to merge it to master in oder to confirm its correct workings and refine the settings (which might be only possible after merge)
@roschaefer @mattwr18

File: .travis.yml [hunk 2]
```diff
@@ -18,17 +20,22 @@ install:
...
   - docker-compose exec backend yarn run test:jest --ci --verbose=false
   - docker-compose exec backend yarn run db:reset
   - docker-compose exec backend yarn run db:seed
   - docker-compose exec backend yarn run test:cucumber
   - docker-compose exec backend yarn run db:reset
   - docker-compose exec backend yarn run db:seed
+  # Frontend
   - docker-compose exec webapp yarn run lint
   - docker-compose exec webapp yarn run test --ci --verbose=false
   - docker-compose exec -d backend yarn run test:before:seeder
```
Comment (Tirokk) [inline .travis.yml:34]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-08T13:58:37Z" title="Wednesday, May 8th 2019, 3:58:37 pm +02:00">May 8, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -18,17 +20,22 @@ install:
   - wait-on http://localhost:7474 && docker-compose exec neo4j migrate
 
 script:
+  # Backend
   - docker-compose exec backend yarn run lint
   - docker-compose exec backend yarn run test:jest --ci --verbose=false
   - docker-compose exec backend yarn run db:reset
   - docker-compose exec backend yarn run db:seed
   - docker-compose exec backend yarn run test:cucumber
   - docker-compose exec backend yarn run db:reset
   - docker-compose exec backend yarn run db:seed
+  # Frontend
   - docker-compose exec webapp yarn run lint
   - docker-compose exec webapp yarn run test --ci --verbose=false
   - docker-compose exec -d backend yarn run test:before:seeder
```


That odd line here is by the way a prerequisite of fullstack testing. Cypress needs to setup data. Therefore we use the same method as in the backend tests: We have a backend server with certain permissions disabled to setup data when we need it. This line here starts the seeder/factory server.

If you want to improve the code comment, you can swap lines 35 <=> 34.

Comment (Tirokk) [reply .travis.yml:34]: > [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-08T14:05:03Z" title="Wednesday, May 8th 2019, 4:05:03 pm +02:00">May 8, 2019</time>_

---


Not in scope of this pr

**Review by Mogge — APPROVED** [review]
Great @ulfgebhardt, sorry I didn't get this completed myself, but it's great to know we can help pick up each other's slack!! really excited about this getting merged in!!

**Pull request closed without merging**